assets/js/components/KeyMetrics/MetricsSelectionPanel/Metrics.js

```diff
@@ -64,14 +67,28 @@ export default function Metrics( { savedMetrics } ) {
 					return acc;
 				}
 
+				const widget = select( CORE_WIDGETS ).getWidget( metric );
+
+				KEY_METRICS_WIDGETS[ metric ].disconnectedModules = [];
+
+				widget.modules.forEach( ( slug ) => {
+					const module = getModule( slug );
+					if ( ! module?.connected ) {
+						KEY_METRICS_WIDGETS[ metric ].disconnectedModules.push(
+							module?.name
+						);
+					}
+				} );
```

Comment:
This can be a bit more concise and not have to modify the array like that.

```suggestion
				KEY_METRICS_WIDGETS[ metric ].disconnectedModules = widget.modules.reduce( ( acc, slug ) => {
					const module = getModule( slug );
					if ( module?.connected || ! module?.name ) {
						return acc;
					}

					return [ ...acc, module.name ];
				}, [] );
```